It's not possible to have 2 libraries which use async functions

Apply this patch to the errors fixture:
```diff
diff --git a/fixtures/errors/src/lib.rs b/fixtures/errors/src/lib.rs
index 2c60467..990bf26 100644
--- a/fixtures/errors/src/lib.rs
+++ b/fixtures/errors/src/lib.rs
@@ -165,4 +165,10 @@ fn error_timestamp() -> Result<std::time::SystemTime, BoobyTrapError> {
     Err(BoobyTrapError::IceSlip)
 }
 
+// Cannot use UDL here: see https://github.com/mozilla/uniffi-rs/issues/1915
+#[uniffi::export]
+async fn async_error() -> Result<(), BoobyTrapError> {
+    Ok(())
+}
+
 include!(concat!(env!("OUT_DIR"), "/errors.uniffi.rs"));
```

Then `./build.sh`, `./build_bindings.sh` and finally `./test_bindings.sh`:
```
--- FAIL: TestFuturesAlwaysReady (0.00s)
panic: interface conversion: interface {} is chan futures._Ctype_schar, not chan errors._Ctype_schar [recovered]
	panic: interface conversion: interface {} is chan futures._Ctype_schar, not chan errors._Ctype_schar

goroutine 29 [running]:
testing.tRunner.func1.2({0x105153580, 0x140000c2090})
	/usr/local/go/src/testing/testing.go:1631 +0x1c4
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1634 +0x33c
panic({0x105153580?, 0x140000c2090?})
	/usr/local/go/src/runtime/panic.go:770 +0x124
github.com/NordSecurity/uniffi-bindgen-go/binding_tests/generated/errors.uniffiFutureContinuationCallbackerrors(0x10?, 0x0)
	.../uniffi-bindgen-go/binding_tests/generated/errors/errors.go:1824 +0x78
github.com/NordSecurity/uniffi-bindgen-go/binding_tests/generated/futures._Cfunc_ffi_uniffi_futures_rust_future_poll_i8(0x600003bf00c0, 0x140000c4048, 0x1400000e1f8)
	_cgo_gotypes.go:263 +0x30
github.com/NordSecurity/uniffi-bindgen-go/binding_tests/generated/futures.AlwaysReady.func2.1(0x600003bf00c0, 0x140000c4048, 0x1400000e1f8)
	.../uniffi-bindgen-go/binding_tests/generated/futures/futures.go:1318 +0x8c
```

The panic is in this function:
```go
func uniffiFutureContinuationCallbackerrors(ptr unsafe.Pointer, pollResult C.int8_t) {
	doneHandle := *(*cgo.Handle)(ptr)
	done := doneHandle.Value().((chan C.int8_t))
	done <- pollResult
}
```

The problem is similar to https://github.com/NordSecurity/uniffi-bindgen-go/issues/43 in that when the runtime type cast is made, it fails because the callback is called with `C.int8_t` _from the `futures` package_, whereas this is type-casting to `C.int8_t` _from the `errors` package_.

In reality, the callback should not be triggered at all from the `errors` package (as the diff doesn't actually call the defined function anywhere) but this happens because it is set when the package is loaded:
```go
func init() {
	uniffiInitContinuationCallback()
	uniffiCheckChecksums()
}
```
This init function appears in both the `errors` and `futures` package, and I guess they step on each other (errors runs first so I guess the 2nd call to  `uniffiInitContinuationCallback()` is a no-op, causing the errors callback to be called for futures tests.

--- a/bindgen/templates/Async.go
+++ b/bindgen/templates/Async.go
@@ -3,8 +3,8 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */#}
 
 const (
-	uniffiRustFuturePollReady      C.int8_t = 0
-	uniffiRustFuturePollMaybeReady C.int8_t = 1
+	uniffiRustFuturePollReady      int8 = 0
+	uniffiRustFuturePollMaybeReady int8 = 1
 )
 
 func uniffiRustCallAsync(
@@ -111,8 +111,8 @@ func uniffiRustCallAsyncInner(
 	pollFunc func(*C.void, unsafe.Pointer, *C.RustCallStatus),
 	freeFunc func(*C.void, *C.RustCallStatus),
 ) (*C.void, error) {
-	pollResult := C.int8_t(-1)
-	waiter := make(chan C.int8_t, 1)
+	pollResult := int8(-1)
+	waiter := make(chan int8, 1)
 	chanHandle := cgo.NewHandle(waiter)
 
 	rustFuture, err := rustCallWithError(converter, func(status *C.RustCallStatus) *C.void {
@@ -146,8 +146,8 @@ func uniffiRustCallAsyncInner(
 //export uniffiFutureContinuationCallback{{ config.package_name.as_ref().unwrap() }}
 func uniffiFutureContinuationCallback{{ config.package_name.as_ref().unwrap() }}(ptr unsafe.Pointer, pollResult C.int8_t) {
 	doneHandle := *(*cgo.Handle)(ptr)
-	done := doneHandle.Value().((chan C.int8_t))
-	done <- pollResult
+	done := doneHandle.Value().((chan int8))
+	done <- int8(pollResult)
 }
 
 func uniffiInitContinuationCallback() {

--- a/binding_tests/issue45_test.go
+++ b/binding_tests/issue45_test.go
@@ -1,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package binding_tests
+
+import (
+	"github.com/NordSecurity/uniffi-bindgen-go/binding_tests/generated/issue45"
+	"testing"
+	"github.com/stretchr/testify/assert"
+)
+
+// Ensure multiple futures packages work fine together, the other one being
+// the "futures" fixture from uniffi-rs.
+// https://github.com/NordSecurity/uniffi-bindgen-go/issues/45
+
+func TestIssue45(t *testing.T) {
+	record := issue45.GetAsyncRecord()
+	assert.Equal(t, record.Id, "foo")
+	assert.Equal(t, record.Tag, "bar")
+}

--- a/fixtures/Cargo.toml
+++ b/fixtures/Cargo.toml
@@ -41,3 +41,4 @@ uniffi-go-fixture-destroy = { path = "destroy" }
 uniffi-go-fixture-errors = { path = "errors" }
 uniffi-go-fixture-name-case = { path = "name-case" }
 uniffi-go-fixture-objects = { path = "objects" }
+uniffi-go-fixture-issue45 = { path = "regressions/issue45" }

--- a/fixtures/regressions/issue45/Cargo.toml
+++ b/fixtures/regressions/issue45/Cargo.toml
@@ -1,0 +1,16 @@
+[package]
+name = "uniffi-go-fixture-issue45"
+version = "1.0.0"
+edition = "2021"
+publish = false
+
+[lib]
+crate-type = ["lib", "cdylib"]
+name = "uniffi_go_issue45"
+
+[dependencies]
+uniffi = {path = "../../../3rd-party/uniffi-rs/uniffi"}
+uniffi_macros = {path = "../../../3rd-party/uniffi-rs/uniffi_macros"}
+
+[build-dependencies]
+uniffi_build = {path = "../../../3rd-party/uniffi-rs/uniffi_build", features=["builtin-bindgen"]}

--- a/fixtures/regressions/issue45/build.rs
+++ b/fixtures/regressions/issue45/build.rs
@@ -1,0 +1,7 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+fn main() {
+    uniffi_build::generate_scaffolding("./src/issue45.udl").unwrap();
+}

--- a/fixtures/regressions/issue45/src/issue45.udl
+++ b/fixtures/regressions/issue45/src/issue45.udl
@@ -1,0 +1,1 @@
+namespace issue45 {};

--- a/fixtures/regressions/issue45/src/lib.rs
+++ b/fixtures/regressions/issue45/src/lib.rs
@@ -1,0 +1,22 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#[derive(uniffi::Record)]
+struct Record {
+    id: String,
+    tag: String,
+}
+
+// Ensure multiple futures packages work fine together, the other one being
+// the "futures" fixture from uniffi-rs.
+// https://github.com/NordSecurity/uniffi-bindgen-go/issues/45
+#[uniffi::export]
+async fn get_async_record() -> Record {
+    Record {
+        id: "foo".to_string(),
+        tag: "bar".to_string(),
+    }
+}
+
+include!(concat!(env!("OUT_DIR"), "/issue45.uniffi.rs"));

--- a/fixtures/src/lib.rs
+++ b/fixtures/src/lib.rs
@@ -37,6 +37,7 @@ mod uniffi_fixtures {
     // Go specific
     uniffi_go_destroy::uniffi_reexport_scaffolding!();
     uniffi_go_errors::uniffi_reexport_scaffolding!();
+    uniffi_go_issue45::uniffi_reexport_scaffolding!();
     uniffi_go_name_case::uniffi_reexport_scaffolding!();
     uniffi_go_objects::uniffi_reexport_scaffolding!();
 }

